packages/flutter/lib/src/widgets/selection_gestures.dart

```diff
@@ -0,0 +1,1251 @@
+  // this value will be set to `0`, and a new series will begin.
+  int get consecutiveTapCount => _consecutiveTapCount;
+  // The set of [LogicalKeyboardKey]'s pressed when the most recent [PointerDownEvent]
+  // was tracked in [addAllowedPointer].
+  //
+  // This value defaults to an empty set.
+  //
+  // When the timer between two taps elapses, the recognizer loses the arena, the gesture is cancelled
+  // or the recognizer is disposed of then this value is reset.
+  Set<LogicalKeyboardKey> get keysPressedOnDown => _keysPressedOnDown ?? <LogicalKeyboardKey>{};
+  // Whether the tap drifted past the tolerance defined by [kDoubleTapTouchSlop] in any subsequent
+  // tracked [PointerMoveEvent]'s.
+  //
+  // This value default to false.
+  //
+  // If the tap does drift past the tolerance then we reset all of the tracked state except
+  // the [currentDown], [currentUp], [consecutiveTapCount], and [keysPressedOnDown]. This is because
+  // the [OneSequenceGestureRecognizer] may be handling a gesture that does accept a tap drift past
+  // the tolerance defined by [kDoubleTapTouchSlop], such as a drag, so it may still want access
+  // to the tracked tap state.
+  bool get pastTapTolerance => _pastTapTolerance;
+  // The upper limit for the [consecutiveTapCount]. When this limit is reached
+  // all tap related state is reset and a new tap series is tracked.
+  //
+  // If this value is null, [consecutiveTapCount] can grow infinitely large.
+  int? get upperLimit;
+
+  // Private tap state tracked.
+  PointerDownEvent? _down;
+  PointerUpEvent? _up;
+  int _consecutiveTapCount = 0;
+  Set<LogicalKeyboardKey>? _keysPressedOnDown;
+  bool _pastTapTolerance = false;
+
+  bool _wonArena = false;
+  OffsetPair? _originPosition;
+  int? _previousButtons;
+
+  // For timing taps.
+  Timer? _consecutiveTapTimer;
+  Offset? _lastTapOffset;
+
+  // When we start to track a tap, we can choose to increment the [consecutiveTapCount]
+  // if the given tap falls under the tolerance specifications or we can reset the count to 1.
+  @override
+  void addAllowedPointer(PointerDownEvent event) {
+    super.addAllowedPointer(event);
+    if (upperLimit != null && upperLimit == _consecutiveTapCount) {
+      _tapTrackerReset();
+    }
+    _up = null;
+    _pastTapTolerance = false;
+    _wonArena = false;
+    if (_down != null && !_representsSameSeries(event)) {
+      // The given tap does not match the specifications of the series of taps being tracked,
+      // reset the tap count and related state.
+      _consecutiveTapCount = 1;
+    } else {
+      _consecutiveTapCount += 1;
+    }
+    _consecutiveTapTimerStop();
+    // `_down` must be assigned in this method instead of [handleEvent],
+    // because [acceptGesture] might be called before [handleEvent],
+    // which may rely on `_down` to initiate a callback.
+    _trackTrap(event);
+  }
+
+  @override
+  void acceptGesture(int pointer) {
+    _wonArena = true;
```

Comment:
I'm not sure I follow. Do you place the `_wonArena` in an else/ after `if` and `if`? 

I think it's fine to set `_wonArena` where we do since `_consecutiveTapTimerStop/Start` don't make use of it. It's mostly if our recognizer declares victory before `handleEvent`, in this case we will not have tracked a `PointerUpEvent` yet so we can't start the timer on `acceptGesture`, so we wait until the `handleEvent` following the acceptance to track our `PointerUpEvent` and start the timer.

Reply:
Sorry that sentence is really confusing now that I re-read it haha.  This is what I meant, though that makes sense and it's totally fine keeping it where it is:

```dart
    if (_up != null && _down != null) {
      _consecutiveTapTimerStop();
      _consecutiveTapTimerStart();
      _wonArena = false;
    } else {
      _wonArena = true;
    }
```

or

```dart
    if (_up != null && _down != null) {
      _consecutiveTapTimerStop();
      _consecutiveTapTimerStart();
      _wonArena = false;
      return;
    }
    _wonArena = true;
```

Reply:
I think I'll leave it as is. 

The first example will only set `_wonArena` to true only if `_up` or `_down` is null, I think this is not accurate because the arena is won at this point regardless if we have an `_up` or `_down`. 

The second example is similar to what I have now, though I think I like setting `_wonArena` to true first.